examples/core_api/timm-post/post.md

```diff
@@ -0,0 +1,286 @@
+(This post references unreleased features, namely https://github.com/determined-ai/determined/pull/3859 and https://github.com/determined-ai/determined/pull/3807.  To run examples, you'll need to create a local branch with those two PRs merged.)
+
+(TODO: Replace github references with doc references where possible.)
+
+(TODO: Are the initial paragraphs the right story to tell about Core API?  Could completely replace them.)
+
+Users of Determined have always faced a small but annoying barrier to entry - porting their model training scripts to use one of Determined's Trial APIs.  Similar to popular libraries like PyTorch Lightning, our Trial APIs provide user-definable hooks for each step in a standard model training workflow.  Our behind-the-scenes execution harness then handles the messy details of providing scalable, distributed, preemptible training with full access to features like hyperparameter search and checkpoint management.
+
+Over time, we've realized that ML teams often have extensive script libraries and custom execution harnesses of their own that make this porting process non-trivial.  To address this, we designed *Core API*: an API with fewer built-in assumptions that can be plugged into existing code with minimal fuss.  We're releasing it as part of Determined 0.17.XX.
+
+In this post, we'll walk through converting an existing training script to use the full set of Core API features.  We'll proceed in steps, to illustrate how Core API enables incremental adoption of features.  And instead of working with a toy example, we'll instead use the training script from Ross Wightman's popular `timm`  library for image models.  TODO: More background on `timm`?
```

Comment:
For the toy example, you might as well link to the new official core api tutorial, which just landed this week.  See docs/training-apis/core-api/*